core/lib/dal/src/consensus_dal/mod.rs

```diff
@@ -361,15 +346,42 @@ impl ConsensusDal<'_, '_> {
         .fetch_optional(self.storage)
         .await?
         {
-            return Ok(BlockStoreState {
-                first,
-                last: Some(Last::Final(
-                    zksync_protobuf::serde::Deserialize {
-                        deny_unknown_fields: true,
-                    }
-                    .proto_fmt(row.certificate)?,
-                )),
-            });
+            let d = zksync_protobuf::serde::Deserialize {
+                deny_unknown_fields: true,
+            };
+
+            // First try to use versioned_certificate
+            let cert = row
+                .versioned_certificate
+                .as_ref()
+                .map(|cert| d.proto_fmt(cert))
+                .transpose()?;
+
+            if let Some(cert) = cert {
+                let last = match cert {
+                    BlockCertificate::V1(qc) => Last::FinalV1(qc),
+                    BlockCertificate::V2(qc) => Last::FinalV2(qc),
+                };
+                return Ok(BlockStoreState {
+                    first,
+                    last: Some(last),
+                });
+            }
+
+            // If versioned_certificate is None, try to use certificate
+            // This is for backward compatibility
+            let qc = row
+                .certificate
+                .as_ref()
+                .map(|cert| d.proto_fmt(cert))
+                .transpose()?;
+
+            if let Some(qc) = qc {
+                return Ok(BlockStoreState {
+                    first,
+                    last: Some(Last::FinalV1(qc)),
+                });
+            }
```

Comment:
Would it make sense to add logs for this for easier debug? Feel free to ignore if not

Reply:
Maybe, I'll think about it.